framework/src/main/java/org/checkerframework/framework/util/AtmLubVisitor.java

```diff
@@ -245,12 +245,14 @@ private void lubWildcard(
         visit(type1LowerBound, type2LowerBound, lubLowerBound);
 
         for (AnnotationMirror top : qualifierHierarchy.getTopAnnotations()) {
-            AnnotationMirror glb =
-                    qualifierHierarchy.greatestLowerBound(
-                            type1LowerBound,
-                            type2LowerBound,
-                            type1LowerBound.getAnnotationInHierarchy(top),
-                            type2LowerBound.getAnnotationInHierarchy(top));
+            AnnotationMirror anno1 = type1LowerBound.getAnnotationInHierarchy(top);
+            AnnotationMirror anno2 = type2LowerBound.getAnnotationInHierarchy(top);
+
+            AnnotationMirror glb = null;
+            if (anno1 != null && anno2 != null) {
```

Comment:
Is this equivalent and a bit clearer:

```java
if (anno1 != null && anno2 != null) {
    AnnotationMirror glb = qualifierHierarchy.greatestLowerBound(anno1, anno2);
    lubLowerBound.replaceAnnotation(glb);
}
```

As anno1 and anno2 are from the same hierarchy, their glb must exist. The two ifs are confusing the logic for me.